staging/src/k8s.io/kubectl/pkg/cmd/apply/apply.go

```diff
@@ -384,6 +384,11 @@ func (o *ApplyOptions) Validate() error {
 		return fmt.Errorf("--force cannot be used with --prune")
 	}
 
+	// Currently do not support pruning objects which are server-side applied.
+	if o.Prune && o.ServerSideApply {
+		return fmt.Errorf("--prune is not supported with server-side apply")
```

Comment:
How about: "objects created by server-side apply cannot currently be pruned because of <issue link>"?

Reply:
Or perhaps "A version of prune that works on objects created by server-side apply is not yet available. Please omit the --prune flag until then."

Reply:
I've attempted a new error message--let me know what you think.